### ISSUE TITLE
Frequent force closes

### ISSUE THREAD

**Issue by aspiers** [EddieRingle/hubroid#49]
With 2.0 beta 3 I get force closes in many situations. For example, start the app, tap Activity Feeds -> immediate FC. This is with a Nexus S. I always click the button to send a report so you should have the stack traces, but I thought it would be worth submitting an issue to track it.

**Comment by EddieRingle** [comment]
From the Market interface I see force closes in the Activity Feeds being caused by a failure to parse Gravatar URLs. If you can, try deleting the hubroid/ directory on your SD card and see what happens (that's where Hubroid caches gravatar IDs and images).

**Comment by aspiers** [comment]
You're right, I see the gravatar references in the stack trace. Removing hubroid from the SD doesn't help, but It's an NPE so presumably it's easy to fix?

**Comment by aspiers** [comment]
Hubroid also FCs when submitting these comments! but for a different reason...

**Comment by EddieRingle** [comment]
Well, the NPE is caused by bad data from the API, so unfortunately it is very had to diagnose.

**Comment by EddieRingle** [comment]
And the above comment was posting via Hubroid on my G2. It's strange that I'm unable to reproduce most of these crashes.

**Comment by aspiers** [comment]
Presumably it's possible to defensively program against NPEs due to the API returning bad data though, so that Hubroid at least recovers gracefully?

Given that I can reproduce these FCs reliably, please let me know if I can help debug them somehow.  I had assumed that the stack traces would be plenty enough info, but if not I could install an .apk of a version with extra debugging in, for instance?

**Comment by aspiers** [comment]
Now I'm getting a FC immediately every time I start the app. It's something to do with JSON. I've sent in the FC report.

**Comment by EddieRingle** [comment]
Once I get some time, I'll release a version that sends me more information when the app crashes (like the JSON involved). This sounds like inconsistencies in the API, which I've encountered before and have caused random issues like this.

**Comment by EddieRingle** [comment]
This should be fixed, actually. Let me know if you experience the issue (new version has been on the Market for a few hours).

**Comment by aspiers** [comment]
OK, now I'm completely confused.  When I originally reported this, I thought I checked what version I had and that it was 2.0 beta 3.  However now I see that it was 1.3.2!  I don't know why the Market app didn't offer me an upgrade to 2.0 ages ago but I think when checking the version I must have incorrectly assumed I was on the latest version and obtained the number from github or market.android.com or something.  So it appears all the FCs I reported are related to 1.3.2 :-(  I have just installed 2.0 beta 3 and am not getting FCs any more.  Sorry for the noise :-(

**Comment by EddieRingle** [comment]
Hehe, that's actually a bad on my part. I had a disk failure awhile ago and lost the key I used to sign the old Hubroid packages. As a result I had to use a different package name and therefore most users never realized there was an updated version available. I've started to back up those keys now, so this shouldn't happen in the future.

Thanks for using Hubroid and taking the time to report issues. :)
